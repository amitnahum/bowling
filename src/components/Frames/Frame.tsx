import React from 'react';
import { FrameCell } from "./FrameCell";


export interface IFrame {
    cells: (string | number)[]
    footer: string | number,
    header: string | number
}

export function Frame({header, footer = '', cells = ['', '']}: IFrame) {
    return (
        <div className={"frame"}>
            <div className={"frame-header"}>{header}</div>
            <div className={"frame-body"}>
                {
                    cells.map((contents: string | number, index) =>
                        <FrameCell key={index} contents={contents}/>)
                }
            </div>
            <div className={"frame-footer"}>{footer}</div>
        </div>
    );
}
