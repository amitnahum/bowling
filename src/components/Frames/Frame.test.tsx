import React from 'react';
import { Frame } from "./Frame";
import renderer from 'react-test-renderer';

describe("Frame Unit Test", function () {
    it('renders correctly with two cells', function () {
        const frame = renderer.create(<Frame cells={[5,5]} footer={10} header={1}/>).toJSON();
        expect(frame).toMatchSnapshot();
    });
    it('renders correctly with three cells', function () {
        const frame = renderer.create(<Frame header={5} cells={['','','']}
                                             footer={10}/>).toJSON();
        expect(frame).toMatchSnapshot();
    });
});
