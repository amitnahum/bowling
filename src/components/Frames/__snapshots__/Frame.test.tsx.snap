// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame Unit Test renders correctly with three cells 1`] = `
<div
  className="frame"
>
  <div
    className="frame-header"
  >
    5
  </div>
  <div
    className="frame-body"
  >
    <div
      className="frame-cell"
    >
      
    </div>
    <div
      className="frame-cell"
    >
      
    </div>
    <div
      className="frame-cell"
    >
      
    </div>
  </div>
  <div
    className="frame-footer"
  >
    10
  </div>
</div>
`;

exports[`Frame Unit Test renders correctly with two cells 1`] = `
<div
  className="frame"
>
  <div
    className="frame-header"
  >
    1
  </div>
  <div
    className="frame-body"
  >
    <div
      className="frame-cell"
    >
      5
    </div>
    <div
      className="frame-cell"
    >
      5
    </div>
  </div>
  <div
    className="frame-footer"
  >
    10
  </div>
</div>
`;
